spec.html

```diff
@@ -28440,6 +28440,209 @@ <h1>
+
+            <emu-note>
+              Synthetic Module Records have no dependencies.
+            </emu-note>
+          </emu-clause>
+
+          <emu-clause id="sec-smr-getexportednames" type="concrete method">
+            <h1>GetExportedNames ( ): a List of Strings</h1>
+            <dl class="header">
+              <dt>for</dt>
+              <dd>a Synthetic Module Record _module_</dd>
+            </dl>
+
+            <emu-alg>
+              1. Return _module_.[[ExportNames]].
+            </emu-alg>
+          </emu-clause>
+
+          <emu-clause id="sec-smr-resolveexport" type="concrete method">
+            <h1>
+              ResolveExport (
+                _exportName_: a String,
+              ): a ResolvedBinding Record or *null*
+            </h1>
+            <dl class="header">
+              <dt>for</dt>
+              <dd>a Synthetic Module Record _module_</dd>
+            </dl>
+
+            <emu-alg>
+              1. If _module_.[[ExportNames]] does not contain _exportName_, return *null*.
+              1. Return ResolvedBinding Record { [[Module]]: _module_, [[BindingName]]: _exportName_ }.
+            </emu-alg>
+          </emu-clause>
+
+          <emu-clause id="sec-smr-Link" type="concrete method">
+            <h1>Link ( ): ~unused~</h1>
+            <dl class="header">
+              <dt>for</dt>
+              <dd>a Synthetic Module Record _module_</dd>
+            </dl>
+
+            <emu-alg>
+              1. Let _realm_ be _module_.[[Realm]].
+              1. Let _env_ be NewModuleEnvironment(_realm_.[[GlobalEnv]]).
+              1. Set _module_.[[Environment]] to _env_.
+              1. For each String _exportName_ in _module_.[[ExportNames]], do
+                1. Perform ! _env_.CreateMutableBinding(_exportName_, *false*).
+                1. Perform ! _env_.InitializeBinding(_exportName_, *undefined*).
+              1. Return ~unused~.
+            </emu-alg>
+          </emu-clause>
+
+          <emu-clause id="sec-smr-Evaluate" type="concrete method">
+            <h1>Evaluate ( ): a Promise</h1>
+            <dl class="header">
+              <dt>for</dt>
+              <dd>a Synthetic Module Record _module_</dd>
+            </dl>
+
+            <emu-alg>
+              1. Let _moduleContext_ be a new ECMAScript code execution context.
+              1. Set the Function of _moduleContext_ to *null*.
+              1. Set the Realm of _moduleContext_ to _module_.[[Realm]].
+              1. Set the ScriptOrModule of _moduleContext_ to _module_.
+              1. Set the VariableEnvironment of _moduleContext_ to _module_.[[Environment]].
+              1. Set the LexicalEnvironment of _moduleContext_ to _module_.[[Environment]].
+              1. Suspend the currently running execution context.
+              1. Push _moduleContext_ on to the execution context stack; _moduleContext_ is now the running execution context.
+              1. Let _steps_ be _module_.[[EvaluationSteps]].
```

Comment:
If we're going to put the result of `[[EvaluationSteps]]` into a promise, as below, then it needs to be constrained to actually return an ES value. Right now there is no such constraint, and the only actual use in the spec is an Abstract Closure which doesn't return anything.

My inclination is to say that in the case of a normal completion the value is not used, and instead we put `undefined` in the Promise.